src/App.js

```diff
@@ -1,16 +1,61 @@
 import React from 'react';
+import { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js';
+
+
+function App () 
+{
+  const [entries, setEntries] = useState(chatMessages);
+
+  const updateHeartStatus = (entryId, updatedHeartStatus) =>
+  {
+    console.log('updateHeartStatus called');
+    const newEntriesList = [];
+    for (const entry of entries)
+    {
+      if (entry.id !== entryId)
+      {
+        newEntriesList.push(entry);
+      } else
+      {
+        const newEntry = {
+          ...entry,
+          liked: updatedHeartStatus,
+        };
+        newEntriesList.push(newEntry);
+      }
+    }
```

Comment:
This works well! As an alternative, can you consider how you would do this with a map?